ips/src/tests.rs

```diff
@@ -200,12 +219,73 @@ fn destroy_should_fail() {
         assert_eq!(
             IpsStorage::<Runtime>::get(0),
             Some(IpsInfoOf::<Runtime> {
-                owner: primitives::utils::multi_account_id::<Runtime, <Runtime as Config>::IpsId>(
-                    0
-                ),
+                parentage: Parentage::Parent(primitives::utils::multi_account_id::<
+                    Runtime,
+                    <Runtime as Config>::IpsId,
+                >(0, None)),
                 metadata: MOCK_METADATA.to_vec().try_into().unwrap(),
-                data: vec![0]
+                data: vec![AnyId::IpfId(0)].try_into().unwrap(),
+                allow_replica: true,
+                ips_type: IpsType::Normal,
             })
         );
     });
 }
+
+#[test]
+fn create_replica_should_work() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(Ipf::mint(
+            Origin::signed(ALICE),
+            MOCK_METADATA.to_vec(),
+            H256::from(MOCK_DATA)
+        ));
+
+        assert_eq!(Ips::next_ips_id(), 0);
+        assert_ok!(Ips::create_ips(
+            Origin::signed(ALICE),
+            MOCK_METADATA.to_vec(),
+            vec![0],
+            true,
+        ));
```

Comment:
After this, please assert that the ips that you created does have the allow_replica field set to true, otherwise this test could pass if somehow that field initialized to false and the create_replica function was broken.

Reply:
Resolved on aa5ae41